app/auth/components/Main.js

```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import { Switch, Route, Redirect } from 'react-router-dom';
+
+import SignUp from './SignUp';
+import Login from './Login';
+
+const Main = () => (
```

Comment:
There is not actually any thing related to component being done here. Only the routes are being configured. So a file named `routes.js` like [this](https://github.com/kilkelly/react-passport-redux-example/blob/master/src/client/routes.js), I think would be more appropriate. Not only the name `main.js` is very generic and not elaborate, but also we would need a function like `requireAuth` in future like in the example I linked earlier.

Reply:
In that case, wouldn't it get complicated if we want to use routes in other components e.g. in the `Navbar` where I have used them? [This](https://medium.com/@dkerrious/react-router-v4-1b04ec250c24) article talks about how a single `routes.js` is not required for `react-router v4`. Also the work of the `requireAuth` function in your linked example can be done using `Redirect`.

Reply:
I don't understand. Why would you put `Router` in `Navbar`? You can just use `Link` for that case. I don't see the demerit of keeping all the `routes` together. That is a better code management I think.